### ISSUE TITLE
Audio routing groups not working?

### ISSUE THREAD

**Issue by rugt0r** [microsoftgraph/microsoft-graph-docs#5460]
I'm not clear on what audio routing groups actually do. I was hoping we would be able to have a call between three participants (A, B and C) and have a situation where A and B can hear one another and B and C can hear one another, but A and C cannot. Is this possible? It would be useful in a contact centre coaching scenario.

During my investigation of this, I encountered the following issue. If my bot calls into a meeting, I can create an audio routing group with the following POST:
https://graph.microsoft.com/beta/app/calls/451f0800-ffff-414b-bbb8-13f5efc5e86b/audioRoutingGroups

I can see by the response that the ID of this routing group is OneToOne. I can request this group thusly:
https://graph.microsoft.com/beta/app/calls/451f0800-ffff-414b-bbb8-13f5efc5e86b/audioRoutingGroups/OneToOne

The documentation suggests I can list the audio routing groups by calling beta/app/calls/{id}/audioRoutingGroups, but doing this returns a 404:
https://graph.microsoft.com/beta/app/calls/451f0800-ffff-414b-bbb8-13f5efc5e86b/audioRoutingGroups

The response contains:
No HTTP resource was found that matches the request URI 
'https://b-pma-euno-01.plat.skype.com:8005/newPlatform/v1/app/calls/451f0800-ffff-414b-bbb8-13f5efc5e86b/audioRoutingGroups'

Is this not implemented? Or am I doing something wrong?

Are we able to use audio routing groups on calls that were escalated from p2p to conferences by inviting a third participant? Or is it just available in scheduled meetings?

Also why does https://docs.microsoft.com/en-us/graph/api/call-post-audioroutinggroups?view=graph-rest-beta contain an example where an audio routing group is created by passing in an ID of "oneToOne" but https://docs.microsoft.com/en-us/graph/api/resources/audioroutinggroup?view=graph-rest-beta states that the id property is "Read-only. Server generated." ?

Thanks.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 7c2befd1-3199-7011-92ca-a7687fd6d434
* Version Independent ID: 402a085f-66e9-227e-ae98-4d378e50b90e
* Content: [List audio routing groups - Microsoft Graph beta](https://docs.microsoft.com/en-us/graph/api/call-list-audioroutinggroups?view=graph-rest-beta&tabs=http)
* Content Source: [api-reference/beta/api/call-list-audioroutinggroups.md](https://github.com/microsoftgraph/microsoft-graph-docs/blob/master/api-reference/beta/api/call-list-audioroutinggroups.md)
* Product: **microsoft-teams**
* Technology: **microsoft-graph**
* GitHub Login: @VinodRavichandran
* Microsoft Alias: **MSGraphDocsVteam**

**Comment by VinodRavichandran** [comment]
@ssulzer Is this something you can help with?

**Comment by ssulzer** [comment]
@VinodRavichandran 
Vinod, I am not directly involved with the design/implementation of audio routing groups in Teams meetings, I will have to consult with some colleagues, and also check with Graph SDK folks, regarding what the intended use case is for audio groups. My understanding is that the feature allows a bot to customize the routing of audio sent to and received from the bot. I do not think it is intended to allow a bot to control whether two other participants in the conversation can hear each other. (That may be a valid feature request to support Contact Center/"Customer Care" scenarios, but I don't think the platform has support for it right now.)

cc: @ksikorsk 

Tanks,
Stephen

**Comment by ksikorsk** [comment]
That is correct @ssulzer.  AudioRoutingGroup functionality is actually quite limited.

You can either create a:
a) one to one bi-directional channel between the bot and a participant
For this scenario specify a `OneToOne` routing mode with a single receiver (sender is ignored).

b) you can created a one to many single-directional blast where bot is the source.
For this scenario specify a `Multicast` routing mode with N receivers (sender is ignored).

Keep in mind that in order for this to take effect, but must join the meeting without being added to the default audio routing group (I.E. `call.MediaConfig.RemoveFromDefaultAudioGroup = false;`)

**Comment by ksikorsk** [comment]
@rugt0r couple answers:
- It should not matter what your media config is.
- The GET groups endpoint does not work (bug). 
  `GET https://graph.microsoft.com/beta/app/calls/451f0800-ffff-414b-bbb8-13f5efc5e86b/audioRoutingGroups`
- The documentation is not in sync with how this API functions, and requires a refresh.

@VinodRavichandran can the issues above be tracked on PMA side?

**Comment by VinodRavichandran** [comment]
Will need inputs from @zhengni-msft

**Comment by rugt0r** [comment]
Thanks for the responses.

>  I do not think it is intended to allow a bot to control whether two other participants in the conversation can hear each other.

This should probably be made more clear in the docs, to save other people from making my mistake. The opening line of the documentation says "The audio routing group stores a private audio route between participants in a multiparty conversation" which prompted me to spend my time on this.

**Comment by WeilinYu2020** [comment]
Hi @ksikorsk , is there any plan or ETA on allowing bot to manipulate audio routing between the participants for p2p/multiparty and meeting call? Thanks.

**Comment by peterdownunder** [comment]
Any update on this. We are interested in using this on groupcall/multiparty calls

**Comment by ssulzer** [comment]
@jsandoval-msft
